pkg/platform/secret.go

```diff
@@ -105,7 +106,9 @@ func ApplyPlatformSecret(
 	if err != nil && !kerrors.IsNotFound(err) {
 		return fmt.Errorf("error getting platform secret %s/%s: %w", namespace, DefaultPlatformSecretName, err)
 	} else if kerrors.IsNotFound(err) {
-		_, err = kubeClient.CoreV1().Secrets(namespace).Create(ctx, &corev1.Secret{
+		// try to find vCluster service
+		vClusterService, svcErr := kubeClient.CoreV1().Services(namespace).Get(ctx, name, metav1.GetOptions{})
+		secret := &corev1.Secret{
```

Comment:
nit: maybe it would be better to first instantiate the secret struct and then retrieve the service. It would make the code a bit more readable and it's also a good practise handling error right after they occur.

Reply:
good point, moved